### PR TITLE
fixed issue with packets being discarded when using multiple universe…

### PR DESCRIPTION
**Pull request by atoudam**
…s (in server mode) due to only a single global lastSequenceNumber being stored, now a separate lastSequenceNumber is stored for each universe

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -38,9 +38,12 @@ function Server(universes, port) {
   this.universes = universes || [0x01];
   this.port = port || e131.DEFAULT_PORT;
   this._socket = dgram.createSocket('udp4');
-  this._lastSequenceNumber = 0;
-
+  this._lastSequenceNumber = {}
```
Comment (hhromic) [inline lib/server.js]: I know this is valid JS, but for consistency please add a terminating `;`

File: lib/server.js [hunk 2]
```diff
@@ -57,14 +60,16 @@ function Server(universes, port) {
...
       self.emit('packet-out-of-order', packet);
+	  
     } else {
       self.emit('packet', packet);
     }
-    self._lastSequenceNumber = packet.getSequenceNumber();
+    self._lastSequenceNumber[packet.getUniverse()] = packet.getSequenceNumber();
   });
   this._socket.bind(this.port, function onListening() {
+
```
Comment (hhromic) [inline lib/server.js]: Please remove this unnecessary blank line

File: lib/server.js (hunk 1 above)
Comment (hhromic) [reply lib/server.js]: also, please also keep the extra blank line that was there before.

**Comment by atoudam** [comment]
no worries, happy to help, love your work on this library it has been very useful 
hopefully i have made those changes correctly, I'm a bit new to the pull request system

File: lib/server.js [hunk 3]
```diff
@@ -38,9 +38,12 @@ function Server(universes, port) {
   this.universes = universes || [0x01];
   this.port = port || e131.DEFAULT_PORT;
   this._socket = dgram.createSocket('udp4');
-  this._lastSequenceNumber = 0;
-
+  this._lastSequenceNumber = {}
   var self = this;
+  self.universes.forEach(function (universe) {
+  	self._lastSequenceNumber[universe] = 0;
+	});
```
Comment (hhromic) [inline lib/server.js]: Please check indentation. I use two spaces only (not tabs), also the `});` bit should be aligned to the beginning `self.universes.forEach(` part.

File: lib/server.js [hunk 4]
```diff
@@ -57,14 +60,16 @@ function Server(universes, port) {
       self.emit('packet-error', packet, validation);
       return;
     }
-    if (packet.discard(self._lastSequenceNumber)) {
+    if (packet.discard(self._lastSequenceNumber[packet.getUniverse()])) {
       self.emit('packet-out-of-order', packet);
+
```
Comment (hhromic) [inline lib/server.js]: Please remove this extra line.

File: lib/server.js [hunk 5]
```diff
@@ -38,9 +38,13 @@ function Server(universes, port) {
   this.universes = universes || [0x01];
   this.port = port || e131.DEFAULT_PORT;
   this._socket = dgram.createSocket('udp4');
-  this._lastSequenceNumber = 0;
-
+  this._lastSequenceNumber = {};
+
```
Comment (hhromic) [inline lib/server.js]: Remove the indentation, but leave the extra line. If you check the diff, you will see you have orphan extra spaces there.

**Comment by hhromic** [comment]
Thanks for the update. Don't worry, we were all learners at some point!
**Tip:** don't close the PR yourself, it will be automatically closed once it's merged.
I found some extra minor coding style issues, review them and I think will be ready to merge.
Thanks again!

**Comment by atoudam** [comment]
thanks, i should have got them all, however i have difficultly confirming i removed the indent but left the line

**Comment by hhromic** [comment]
Hi, sorry for the long delay on this. I got very busy recently and completely forgot about this PR. I will merge it now. Thanks for the contribution and the fix!

**Pull request merged**